tools/opam/process.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/sh
+#**************************************************************************
+#*                                                                        *
+#*                                 OCaml                                  *
+#*                                                                        *
+#*            David Allsopp, University of Cambridge & Tarides            *
+#*                                                                        *
+#*   Copyright 2025 David Allsopp Ltd.                                    *
+#*                                                                        *
+#*   All rights reserved.  This file is distributed under the terms of    *
+#*   the GNU Lesser General Public License version 2.1, with the          *
+#*   special exception on linking described in the file LICENSE.          *
+#*                                                                        *
+#**************************************************************************
+
+set -e
```

Comment:
for some extra safety maybe?
```suggestion
set -eu
if set -o pipefail > /dev/null 2> /dev/null; then
  # POSIX.1-2024
  set -o pipefail
fi
```

Reply:
same question for the generated `-clone` and `-fixup` scripts

Reply:
Yes, indeed - that's a good idea!

Reply:
What am I missing in calling it twice? Isn't

```sh
set -o pipefail > /dev/null 2> /dev/null || true
```

sufficient?

Reply:
oh yes, of course

Reply:
Hah - it turns out neither of us appears to have been correct! Apparently shell builtins are allowed to abort execution unconditionally, so it appears to be necessary to have:

```sh
if (set -o pipefail 2>/dev/null); then
  set -o pipefail
fi
```

With Ubuntu's default dash, both this:

```sh
#!/bin/sh

set -o pipefail || echo No pipefail
```
and
```sh
#!/bin/sh

if ! set -o pipefail; then
  echo No pipefail
fi
```
display an error from the `set` command but then immediately abort the script.